services/Tel.Egram.Utils/TdLib/IAgent.cs

```diff
@@ -7,6 +7,8 @@ public interface IAgent
     {
         IObservable<TdApi.Update> Updates { get; }
 
+        IObservable<TdApi.ConnectionState> ObserveConnectionState();
```

Comment:
Agent isn't meant to be used directly. It's purpose is to execute actions and receive updates. Instead the preferred approach is to create a wrapper around it. You could see how it's done with IAuthenticator for example.